File: src/main/java/mediathek/gui/actions/DeleteDownloadAction.java
package mediathek.gui.actions;

import mediathek.gui.tabs.tab_downloads.GuiDownloads;
import mediathek.tool.SVGIconUtilities;

import javax.swing.*;
import java.awt.event.ActionEvent;

public class DeleteDownloadAction extends AbstractAction {
    private final GuiDownloads guiDownloads;

    public DeleteDownloadAction(GuiDownloads guiDownloads) {
        this.guiDownloads = guiDownloads;
        putValue(Action.SMALL_ICON, SVGIconUtilities.createSVGIcon("icons/fontawesome/xmark.svg"));
        putValue(Action.NAME, "gespeicherten Film (Datei) löschen");
    }

    @Override
    public void actionPerformed(ActionEvent e) {
        guiDownloads.filmLoeschen_();
    }
}

File: src/main/kotlin/mediathek/gui/actions/DeleteDownloadAction.kt
package mediathek.gui.actions

import mediathek.gui.tabs.tab_downloads.GuiDownloads
import mediathek.tool.SVGIconUtilities
import java.awt.event.ActionEvent
import javax.swing.AbstractAction

class DeleteDownloadAction(private val guiDownloads: GuiDownloads) : AbstractAction() {
    init {
        putValue(SMALL_ICON, SVGIconUtilities.createSVGIcon("icons/fontawesome/xmark.svg"))
        putValue(NAME, "Gespeicherten Film (Datei) löschen")
    }

    override fun actionPerformed(e: ActionEvent?) {
        guiDownloads.deleteDownloadedFile()
    }
}
